src/binder/clrprivbinderassemblyloadcontext.cpp

```diff
@@ -240,6 +255,15 @@ HRESULT CLRPrivBinderAssemblyLoadContext::SetupContext(DWORD      dwAppDomainId,
                 // AssemblyLoadContext instance
                 pBinder->m_ptrManagedAssemblyLoadContext = ptrAssemblyLoadContext;
 
+                // Link to LoaderAllocator, keep a reference to it
+                VERIFY(pLoaderAllocator->AddReferenceIfAlive());
+                pBinder->m_pAssemblyLoaderAllocator = pLoaderAllocator;
+                pBinder->m_loaderAllocatorHandle = loaderAllocatorHandle;
+
+#if !defined(DACCESS_COMPILE) && !defined(CROSSGEN_COMPILE)
+                ((AssemblyLoaderAllocator*)pLoaderAllocator)->SetBinderToRelease(pBinder);
```

Comment:
I would suggest naming "SetBinderToRelease" to something like "RegisterBinder" since this is about setup a 1:1 association between LoaderAllocator and ALC.